savoirfairelinux/sous-chef
# Fixed #386: note list view - multiple "mark as read/unread" in one click

## Fixes #386  by lingxiaoyang

### Changes proposed in this pull request:

* Emphasize unread notes
* Change mark as read/unread buttons as "checkboxes".
* Provide "Commit selected changes" and "Mark all as read" buttons.
* In backend, create new batch toggle view to handle this request. (unit test included)

![image](https://cloud.githubusercontent.com/assets/8630726/22001383/c9a6185a-dc12-11e6-9fc0-bf40e2f48b39.png)
![image](https://cloud.githubusercontent.com/assets/8630726/22001443/4428d0ae-dc13-11e6-9238-c4273debd02f.png)


### Status


### How to verify this change

The new functionality concerns two pages:

1. Note list view: http://localhost:8000/note/
2. Client note view: http://localhost:8000/member/view/915/notes

Originally, clicking mark as read/unread button should trigger an update and a page refresh immediately.

Now, these buttons work like "checkboxes". It is possible to select multiple of them and commit at once.

### Deployment notes and migration

none

### New translatable strings


### Additional notes

**Needs review on the design of UI/UX**
